src/macaron/database/rfc3339_datetime.py

```diff
@@ -14,34 +14,33 @@ class RFC3339DateTime(TypeDecorator):  # pylint: disable=W0223
     SQLAlchemy column type to serialise datetime objects for SQLite in consistent format matching in-toto.
 
     https://docs.sqlalchemy.org/en/20/core/custom_types.html#store-timezone-aware-timestamps-as-timezone-naive-utc
+    https://docs.sqlalchemy.org/en/20/dialects/sqlite.html#sqlalchemy.dialects.sqlite.DATETIME
     """
 
     # It is stored in the database as a string
     impl = String
     cache_ok = True
 
-    def process_bind_param(self, value: Optional[Any], dialect: Any) -> None | str:
-        """Process when storing.
+    def process_bind_param(self, value: None | Any, dialect: Any) -> None | str:
+        """Process when storing. If the provided ``datetime`` is a naive datetime object then UTC is added.
 
         value: None | datetime.datetime
             The value being stored
         """
-        result = None
-        if value is not None:
-            if not value.tzinfo:
-                raise TypeError("tzinfo is required")
-            value = value.astimezone(datetime.timezone.utc).replace(tzinfo=None)
-            result = value.isoformat(sep="T", timespec="seconds") + "Z"
-        return result
-
-    def process_result_value(self, value: Optional[Any], dialect: Any) -> None | datetime.datetime:
+        if value is None:
+            return None
+        if not isinstance(value, datetime.datetime):
+            raise TypeError("RFC3339DateTime type expects a datetime object")
+        if not value.tzinfo:
+            value = value.astimezone(datetime.timezone.utc)  # Consider coercing to host timezone.
```

Comment:
Can you please make this comment a TODO?

Reply:
Do you want me to make the change here, or in a separate PR?

Reply:
I prefer a separate PR.